app/scripts/lib/mmi-hooks.ts

```diff
@@ -0,0 +1,66 @@
+import { TransactionMeta } from '../controllers/transactions/tx-state-manager';
+
+/**
+ * Whether or not to skip publishing the transaction.
+ *
+ * @param txMeta - The transaction meta.
+ * @param signedEthTx - Signed Ethereum transaction.
+ * @param addTransactionToWatchList
+ */
+export function afterTransactionSign(
+  txMeta: TransactionMeta,
+  signedEthTx: any,
+  addTransactionToWatchList: (
+    custodianTransactionId: string | undefined,
+    from?: string,
+    bufferType?: string,
+    isSignedMessage?: boolean,
+  ) => Promise<void>,
+): boolean {
+  // MMI does not broadcast transactions, as that is the responsibility of the custodian
+  if (txMeta?.custodyStatus) {
```

Comment:
Very minor, but in cases like this were the bulk of the logic is nested, should we instead use an early return?

```
if (!txMeta?.custodyStatus) {
    return true;
}

txMeta.custodyId = signedEthTx.custodian_transactionId;
txMeta.custodyStatus = signedEthTx.transactionStatus;

addTransactionToWatchList(txMeta.custodyId, txMeta.txParams.from);

return false;
```